src/codegen/general.rs

```diff
@@ -177,6 +177,36 @@ pub fn define_shared_type(
     Ok(())
 }
 
+pub fn cfg_deprecated(
+    w: &mut Write,
+    deprecated: Option<Version>,
+    commented: bool,
+    indent: usize,
+) -> Result<()> {
+    if let Some(s) = cfg_deprecated_string(deprecated, commented, indent) {
+        try!(writeln!(w, "{}", s));
+    }
+    Ok(())
+}
+
+pub fn cfg_deprecated_string(
+    deprecated: Option<Version>,
+    commented: bool,
+    indent: usize,
+) -> Option<String> {
+    if let Some(v) = deprecated {
+        let comment = if commented { "//" } else { "" };
+        Some(format!(
+            "{}{}#[cfg_attr({}, deprecated)]",
```

Comment:
I don't fully understand how this should work. The optional "deprecated-version" attribute in the gir file is for the native library version, so we would have to e.g. generate a `feature = "v2_40"`. Is that what `v.to_cfg()` does?
What for the case where there is no version, we just ignore that case?

I don't see you parsing the deprecated-version field from the gir file, only something from the toml configuration.

Reply:
AFAIU it would work as `#[cfg_attr(feature = "v2_40", deprecated)]`, not just `#[cfg_attr("v2_40", deprecated)]`.

Reply:
`to_cfg` generates `feature = "v2_40"`. Also, I added the parse of the deprecated field [here](https://github.com/gtk-rs/gir/pull/518/files#diff-08faf151bfcc6c456f5db4c647591a76R42). But more globally, we already get the deprecated version when parsing the gir files (when I generated for gtk and gio, it got the information). But maybe I'm wrong or misunderstood you?

Reply:
That line is parsing it from the toml config, not from the gir file. You mean the code to parse it from the gir file was already in src/parser.rs before? All good then

Reply:
Yes, we already got this information before (I didn't add much `deprecated_version` fields as you can see). :)

Reply:
Lucky you :) All good then